### PR TITLE
Lockdown ktlint version

### PR DESCRIPTION
**Pull request by carlosfelix2**
This is just a preemptive lockdown of the ktlint version being used. If the plugin is updated and also updates the ktlint version then we will see a lot ktlint errors.  This change is also meant to allow us to migrate to newer versions of ktlint gradually.  The version was set to the current version used by the plugin (see https://github.com/JLLeitschuh/ktlint-gradle/blob/master/plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintExtension.kt#L38)

Test Steps:
1. Run `./gradlew ktlintcheck` and verify the check is successful.

## Changes
- Lock down the ktlint version to what is used by the gradle plugin.
- Updated getting started doc

### Testing
- [x] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

**Comment by TurkeyFried** [comment]
OH HELL YEAH

**Review by jimmyfagan — APPROVED** [review]
I think this is good, but just as a heads up, the linter only cares about certain formatting issues, it's still happy with or without trailing commas, and it's okay with certain whitespace differences. Still, this is good progress, it does fix some of the linebreak discrepancies. And there may be additional things we can do later in terms of IDE settings to unify some of the other formatting differences.

File: prime-router/build.gradle.kts [hunk 1]
```diff
@@ -581,7 +591,7 @@ jooq {
                                     // A Java regex matching fully-qualified columns, attributes, parameters. Use the pipe to separate several expressions.
                                     // If provided, both "includeExpressions" and "includeTypes" must match.
                                     .withIncludeExpression("action_log.detail")
-                                    .withIncludeTypes("JSONB"),
+                                    .withIncludeTypes("JSONB")
```
Comment (clediggins-usds) [inline prime-router/build.gradle.kts:594]: You and Jimmy appear to have opposite linting conditions for the trailing comma.

**Review by clediggins-usds — APPROVED** [review]

**Comment by carlosfelix2** [comment]
> I think this is good, but just as a heads up, the linter only cares about certain formatting issues, it's still happy with or without trailing commas, and it's okay with certain whitespace differences. Still, this is good progress, it does fix some of the linebreak discrepancies. And there may be additional things we can do later in terms of IDE settings to unify some of the other formatting differences.

That all depends on the ktlint version being used.  Newer versions add those checks you mentioned, but unsure the program wants those at this point.

File: prime-router/build.gradle.kts (hunk 1 above)
Comment (carlosfelix2) [reply prime-router/build.gradle.kts:594]: Yes, different ktlint versions are being used.  I am not using the one set by the gradle plugin, which is used in the build.

**Pull request merged**